tools/blackfire/docker-compose.override.yml

```diff
@@ -0,0 +1,123 @@
+version: '3.7'
```

Comment:
This file should just contain the actual overrides. Docker Compose will merge the overrides with the original config.

See https://docs.docker.com/compose/extends/#understanding-multiple-compose-files and https://docs.docker.com/compose/extends/#adding-and-overriding-configuration

Just adding the actual overrides here makes it easier to digest what it actually adds and makes it more maintainable.

Reply:
We can remove it if you like.

Reply:
For example, here's my local override file that can support both xhprof and blackfire. It's more readable imo:

```yml
services:
  mysql:
    image: mariadb:focal

  xhgui:
    # https://hub.docker.com/r/spacedmonkey/xhgui/tags
    image: spacedmonkey/xhgui:latest
    restart: always
    environment:
      - XHGUI_MONGO_HOSTNAME=mongo
      - XHGUI_MONGO_DATABASE=xhprof
    ports:
      - "8142:80"
    depends_on:
      - mongo
    networks:
      - wpdevnet

  mongo:
    # https://hub.docker.com/r/spacedmonkey/mongodb/tags
    image: spacedmonkey/mongodb:latest
    # (case sensitive) engine: mmapv1, rocksdb, wiredTiger, inMemory
    command: --storageEngine=wiredTiger
    restart: always
    volumes:
      - mongodb:/data/db
    ports:
      - "27017:27017"
    networks:
      - wpdevnet

  blackfire:
    image: blackfire/blackfire:2
    networks:
      - wpdevnet
    ports: ["8307"]
    environment:
      BLACKFIRE_LOG_LEVEL: 4
      BLACKFIRE_SERVER_ID: $BLACKFIRE_SERVER_ID
      BLACKFIRE_SERVER_TOKEN: $BLACKFIRE_SERVER_TOKEN
      BLACKFIRE_CLIENT_ID: $BLACKFIRE_CLIENT_ID
      BLACKFIRE_CLIENT_TOKEN: $BLACKFIRE_CLIENT_TOKEN

  php:
    image: spacedmonkey/php-blackfire:${LOCAL_PHP-latest}
#    image: spacedmonkey/php-xhprof:${LOCAL_PHP-latest}
    environment:
      - BLACKFIRE_CLIENT_ID=${BLACKFIRE_CLIENT_ID-}
      - BLACKFIRE_CLIENT_TOKEN=${BLACKFIRE_CLIENT_TOKEN-}
    depends_on:
      - mysql
      - memcached
      - xhgui

  wordpress-develop:
    depends_on:
    - php
    - blackfire

volumes:
  webroot-share:
  mongodb:
```

Reply:
This would not work, as the docker container for php need to have xhprof installed.

Reply:
@swissspidy Let's do this in a follow PR, as we need to change https://github.com/GoogleChromeLabs/wpp-research/tree/main/tools/xhprof as well.

Reply:
That works for me.

But the override file could still be smaller in this PR.

The whole file can just be:

```yml
services:
  mysql:
    image: mariadb:focal

  blackfire:
    image: blackfire/blackfire:2
    networks:
      - wpdevnet
    ports: ["8307"]
    environment:
      BLACKFIRE_LOG_LEVEL: 4
      BLACKFIRE_SERVER_ID: $BLACKFIRE_SERVER_ID
      BLACKFIRE_SERVER_TOKEN: $BLACKFIRE_SERVER_TOKEN
      BLACKFIRE_CLIENT_ID: $BLACKFIRE_CLIENT_ID
      BLACKFIRE_CLIENT_TOKEN: $BLACKFIRE_CLIENT_TOKEN

  php:
    image: spacedmonkey/php-blackfire:${LOCAL_PHP-latest}
    environment:
      - BLACKFIRE_CLIENT_ID=${BLACKFIRE_CLIENT_ID-}
      - BLACKFIRE_CLIENT_TOKEN=${BLACKFIRE_CLIENT_TOKEN-}
    depends_on:
      - mysql
      - memcached
      - xhgui

  wordpress-develop:
    depends_on:
    - php
    - blackfire

```